### PR TITLE
Feature/use npm workspaces

### PR DESCRIPTION
**Pull request by joswarmer**
I have replaced all shell script by npm commands.  With nmp workspaces it is easy to run an npm command in each package.  This way each package can run their own build/lint/setup/clean/... and it can be run globally.

Also, this is more in line with typical JS/TS projects.
I have used the same setup in lionweb-repository.

### CODE REVIEW

**Comment by joswarmer** [comment]
In addition to this, we can also have `test` scripts per package, that makes is trivial to test each package in isolation (what I have been doing for the last month).

**Review by dslmeinte — CHANGES_REQUESTED** [review]
Three questions/remarks:

* The per-package `README.md` still mention `build.sh`.
* Is there a replacement for the “quick build” functionality? Maybe not necessary though, as this approach seems to run generally a lot quicker than the script-based one!
* Let's put `"typescript": "5.3.3"` as a dev-dependency in the top-level `package.json`?

File: package.json [hunk 1]
```diff
@@ -16,10 +16,21 @@
...
+    "build": "npm run build -w packages --if-present",
+    "lint": "npm run lint -w packages --if-present",
+    "setup": "npm run setup -w packages --if-present",
+    "test": "npm run test -w packages --if-present",
+    "release": "npm run release -w packages --if-present"
   },
   "private": true,
   "workspaces": [
-    "./packages/*"
+    "./packages/core",
```
[the remark this answers is not in the document]
Comment (joswarmer) [reply package.json:29]: Yes, it does. That's why I replaced the * with the actual package names.

File: packages/validation/package.json [hunk 2]
```diff
@@ -14,12 +14,14 @@
...
+    "lint": "eslint src",
+    "release": "npm publish",
     "testValid": "node dist/runners/RunCheckFolder.js ../../../lionweb-integration-testing/testset/valid",
     "testInvalid": "node dist/runners/RunCheckFolder.js ../../../lionweb-integration-testing/testset/invalid",
     "testInvalidWithLanguage": "node dist/runners/RunCheckFolderWithLanguage.js ../../../lionweb-integration-testing/testset/withLanguage/invalid/ ../../../lionweb-integration-testing/testset/withLanguage/myLang.language.json",
     "testValidWithLanguage": "node dist/runners/RunCheckFolderWithLanguage.js ../../../lionweb-integration-testing/testset/withLanguage/valid/ ../../../lionweb-integration-testing/testset/withLanguage/myLang.language.json",
-    "test": "mocha",
     "publish-local": "npm publish --registry http://localhost:4873",
     "unpublish-local": "npm unpublish --force --registry http://localhost:4873",
     "republish-local": "yarn unpublish-local && yarn publish-local"
```
[the remark this answers is not in the document]
Comment (joswarmer) [reply packages/validation/package.json:27]: Since it is empty, yes I will remove it.

**Comment by joswarmer** [comment]
- The per-package README.md still mention build.sh.
  Didnt realize this, will change them.
- Is there a replacement for the “quick build” functionality? Maybe not necessary though, as this approach seems to run generally a lot quicker than the script-based one!
  IS identical to `npm run build` in the root folder
- Let's put "typescript": "5.3.3" as a dev-dependency in the top-level package.json?
   Right, makes it more consistent.

**Pull request merged**